### PR TITLE
[Libcaer] Update to latest version

### PR DESCRIPTION
**Pull request by jureviciusr**
**Describe the pull request**

- #### What does your PR fix?

Updates libcaer port to use latest repository update which incorporates required changes. No need for patch files to install the package.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
yes

### CODE REVIEW

File: ports/libcaer/portfile.cmake [hunk 1]
```diff
@@ -2,19 +2,17 @@ vcpkg_from_gitlab(
...
+	HEAD_REF da6e7cc31a239cf6c071bb110a285f8ebd81c628
 )
 
 vcpkg_cmake_configure(
 	SOURCE_PATH ${SOURCE_PATH}
 	OPTIONS
 		-DENABLE_OPENCV=ON
 		-DEXAMPLES_INSTALL=OFF
-		-DENABLE_BINDIR_INSTALLATION=ON
+		-DBUILD_CONFIG_VCPKG=ON
```
Comment (Osyotr) [inline ports/libcaer/portfile.cmake:15]: Why not just pass correct string into `vcpkg_cmake_config_fixup` and suggest upstream to not hardcode vcpkg-related things into their code?

Comment (jureviciusr) [reply ports/libcaer/portfile.cmake:15]: Thanks for the suggestion, I investigated the option to not apply the `-DBUILD_CONFIG_VCPKG=ON` flag and use the appropriate paths in `vcpkg_cmake_config_fixup`. The install succeeds in that case, but loading the project fails because the fixup method places certain exported files differently and it's not expected by some configuration scripts. 

I will propose to fix this issue and to change the exported cmake scripts that fail to comply explicit flags. Since the library has to support different platforms, such changes will take some time and I would still prefer to keep this flag until a proper solution is implemented and tested on other packaging platforms.

**Review by JackBoosY — DISMISSED** [review]

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/libcaer/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "libcaer",
-  "version": "3.3.14",
-  "port-version": 1,
+  "version": "3.3.14-1",
```
Comment (Thomas1664) [inline ports/libcaer/vcpkg.json:3]: Why did you add `-1` here?

Comment (jureviciusr) [reply ports/libcaer/vcpkg.json]: It was actually unnecessary, I removed it. Thanks for pointing this out.

File: ports/libcaer/libcaer-static-build.patch [hunk 3]
```diff
@@ -1,74 +0,0 @@
...
-+OPTION(BUILD_SHARED_LIBS "Build libcaer as a shared library" ON)
- 
- # Cross-compile support
- IF(NOT USER_LOCAL_PREFIX)
-@@ -50,7 +52,7 @@ MESSAGE(STATUS "Base libraries: ${BASE_LIBS}")
- INCLUDE(FindPkgConfig)
- 
- # Required: basic USB devices support
--IF(CC_MSVC)
-+IF(ENABLE_BINDIR_INSTALLATION)
```
Comment (BillyONeal) [inline ports/libcaer/libcaer-static-build.patch:19]: Can you explain why we no longer need this patch?

Comment (BillyONeal) [reply ports/libcaer/libcaer-static-build.patch:19]: Nevermind, I see this is effectively applied upstream as  https://gitlab.com/inivation/dv/libcaer/-/commit/da6e7cc31a239cf6c071bb110a285f8ebd81c628

File: versions/l-/libcaer.json [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "bd2dbad3c8e09365fd8b15b30afd6bcd5409fee2",
+      "version": "3.3.14",
+      "port-version": 2
+    },
+    {
+      "git-tree": "cb894f482e630e8c983cdee4f3e2e8ecc84a907c",
+      "version": "3.3.14-1",
+      "port-version": 2
+    },
```
Comment (BillyONeal) [inline versions/l-/libcaer.json:12]:
```suggestion
```
A PR must add at most one version database entry. I tried to fix this for you but the PR was opened without permission for us to do so:

image

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: versions/l-/libcaer.json (hunk 4 above)
Comment (jureviciusr) [reply versions/l-/libcaer.json]: I added the suggested change

**Review by JonLiu1993 — APPROVED** [review]

File: ports/libcaer/vcpkg.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "libcaer",
   "version": "3.3.14",
```
Comment (JonLiu1993) [inline ports/libcaer/vcpkg.json:3]:
```suggestion
  "version-date": "2022-07-25",
```

File: ports/libcaer/vcpkg.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
-  "port-version": 1,
+  "port-version": 2,
```
Comment (JonLiu1993) [inline ports/libcaer/vcpkg.json:4]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libcaer/vcpkg.json (hunk 5 above)
Comment (JonLiu1993) [reply ports/libcaer/vcpkg.json:3]: @jureviciusr, Since you updated this port to the latest commit, so here should use `version-date` instead of `version`, see [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/specifications/versioning.md#version-date).

File: ports/libcaer/vcpkg.json (hunk 6 above)
Comment (JonLiu1993) [reply ports/libcaer/vcpkg.json:4]: When we update port version, we should remove `port-version`

**Review by JonLiu1993 — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]
